src/Emulator/Peripherals/Peripherals/Mocks/HPSHostController.cs

```diff
@@ -366,6 +506,13 @@ private void IssueCommand(byte[] data)
             currentSlave.Write(data);
         }
 
+        private bool SocketConnected(Socket s)
+        {
+            bool dataAvailableForReading = s.Poll(1000, SelectMode.SelectRead);
```

Comment:
I think the name of this variable might be misleading - according to the documentation `socket.Poll` might return `true` in other cases too: https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socket.poll?view=netframework-4.7.

Reply:
https://stackoverflow.com/questions/2661764/how-to-check-if-a-socket-is-connected-disconnected-in-c This is where I got this idea from, I agree the wording might be confusing but from my understanding it returns false when the client is completely disconnected, might be different from "closed, reset and terminated". That's also why we have another variable `availableBytesIsEmpty`here too.